### ISSUE TITLE
CPU usage 100%

### ISSUE THREAD

**Issue by kitech** [savoirfairelinux/opendht#224]
full code is here, help check the case:

about send 10 or more several 10 message, the CPU usage will quickly up to 100%

using opendht version: 1.5.0-release

compile:

g++ -g -O2 -std=c++11 tests/cpu100.cpp  -lopendht

```
#include <unistd.h>
#include <opendht.h>


using namespace dht;
static std::mt19937_64 rd {dht::crypto::random_device{}()};
static std::uniform_int_distribution<dht::Value::Id> rand_id;

int main(int argc, char **argv) {

    DhtRunner dht;
    dht.run(0, dht::crypto::generateIdentity("DHT Chat Node"), true, 0);
    const InfoHash myid = dht.getId();
    std::cout<<myid.toString()<<std::endl;
    std::cout<<"port:"<<dht.getBoundPort( )<<std::endl;

    dht.bootstrap("bootstrap.ring.cx", "4222");

    InfoHash room = InfoHash::get(std::string("testroom"));
    std::cout<<"room:"<<"testroom="<<room.toString()<<std::endl;

    int sno = 0;
    while (true) {
        auto now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
        char buf[128]={};
        sprintf(buf, "msg-%d", sno++);
        std::string line = std::string(buf);
        dht.putSigned(room, dht::ImMessage(rand_id(rd), std::move(line), now), [=](bool ok) {
                //dht.cancelPut(room, id);
                if (not ok)
                    std::cout << "Message publishing failed !" << std::string(buf)
                              << "," << std::endl;
            });
        sleep(5);
    }
    dht.join();
    return 0;
}

```

**Comment by AmarOk1412** [comment]
@kitech Should be fixed with OpenDHT 1.6.0 just released. Can you confirmed?

Sorry for the delay!

**Comment by kitech** [comment]
with lastest 1.7 this is not problem anymore.
